### PR TITLE
Select soldiers branch wise on double click.

### PR DESCRIPTION
**Pull request by andreas-eberle**

### CODE REVIEW

**Comment by andreas-eberle** [comment]
@michaelzangl: Can you have a look at the changes?

File: jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java [hunk 1]
```diff
@@ -572,18 +573,31 @@ private IGuiMovable getSelectableMovable(short x, short y) {
...
 			setSelection(new SelectionSet());
 			return;
 		}
 
-		final List<ISelectable> selected = new LinkedList<ISelectable>();
-	
+		EMovableType selectedType = selectedMovable.getMovableType();
+
+		Set<EMovableType> selectableTypes;
+		if (selectedType.isSwordsman()) {
```
Comment (michaelzangl) [inline jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java]: We do not need that case work here. Have a look at the android-selection branch. I improved the ESoldierType. You could add a getSoldierType() method to EMovableType and then compare the soldier type.

Comment (andreas-eberle) [inline jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java]: But what would I do with the normal movables? They cannot have a soldier type, so I would need a special case handling for them...

Comment (michaelzangl) [inline jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java]: Yes, you would and you can still leave the else part. This should just replace the isPikeman()..

You can then get the movable set from the ESoldierType.

Comment (andreas-eberle) [inline jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java]: I think we should stick with this implementation for now. When we extend the implementations of ESoldierType and ESoldierClass, we can use those functions here as well.

**Pull request merged**